### ISSUE TITLE
how can i save user_id to table

### ISSUE THREAD

**Issue by vinnyparker** [crocodic-studio/crudbooster#1257]
Operating system (centos 6.5)
Web server (apache, 64b)
Browser (chrome lastest version)
CRUDBooster version (5.3)

I would like to save user id to a table, to avoid user edit or delete others user`s data.
how can i do that?

**Comment by vinnyparker** [comment]
thanx for answer. 
in this case I use a hook to for save it?

**Comment by andreshg112** [comment]
```php
public function hook_before_add($postdata){
    $postdata['user_id'] = CRUDBooster::myId();
}
```

**Comment by hitmc** [comment]
> Operating system (centos 6.5)
> Web server (apache, 64b)
> Browser (chrome lastest version)
> CRUDBooster version (5.3)
> 
> I would like to save user id to a table, to avoid user edit or delete others user`s data.
> how can i do that?

Hey Vinny, I am new to whole laravel stuffs. just set up the crudbooster. Where i need to insert all these codes. Glad you got it working. Hope you can help me too.

**Comment by vinnyparker** [comment]
> > Operating system (centos 6.5)
> > Web server (apache, 64b)
> > Browser (chrome lastest version)
> > CRUDBooster version (5.3)
> > I would like to save user id to a table, to avoid user edit or delete others user`s data.
> > how can i do that?
> 
> Hey Vinny, I am new to whole laravel stuffs. just set up the crudbooster. Where i need to insert all these codes. Glad you got it working. Hope you can help me too.

you go to /var/www/html/crudbooster/app/Http/Controllers and inside your controller u look for "hook" than put your code there.

**Comment by JulyanoF** [comment]
@vinnyparker did you got working to avoid user edit others user's data?
I'm saving user_id in my table, but when I redirect user or anything else in hook_before_edit, it's editing:
```php
	public function hook_before_edit(&$postdata, $id)
	{
		if(!DB::table('clientes')->where('id', $id)->where('id_cms_users', 2)->first()) {
			return redirect('admin/clientes');
		}
	}

I can't get it work. My condition is valid, but return redirect, return false or anithing else doesn't cancel edition

**Comment by vinnyparker** [comment]
> @vinnyparker did you got working to avoid user edit others user's data? I'm saving user_id in my table, but when I redirect user or anything else in hook_before_edit, it's editing:
> 
> ```
> 	public function hook_before_edit(&$postdata, $id)
> 	{
> 		if(!DB::table('clientes')->where('id', $id)->where('id_cms_users', 2)->first()) {
> 			return redirect('admin/clientes');
> 		}
> 	}
> ```
> 
> I can't get it work. My condition is valid, but return redirect, return false or anithing else doesn't cancel edition

Hi I did it using the privilege and the user_id.
I did it  on hook_query_Index, because, if is admin can see and edit everything, but if is "morador" ( this is a privilege) so I hook to show only the user own data.

`public function hook_query_index(&$query) {
	        //Your code here
	        $user =  CRUDBooster::myId();
	        $privilege = CRUDBooster::myPrivilegeName();
	     
	       if ($privilege =="Morador") {
			   $query = $query->where('cms_user_id', $user);
		   }
		   
	    }
`

**Comment by JulyanoF** [comment]
I was trying to forbid user edit/delete other user data.
I got it overriding parent method:
```php
	public function getEdit($id)
	{
		if (!DB::table('vendas')->where('id', $id)->where('id_cms_users', CRUDBooster::myId())->first()) {
			return redirect('admin/vendas')
				->with([
					'message'      => 'Você não tem permissão para editar essa venda',
					'message_type' => 'danger',
				]);
		}
		return parent::getEdit($id);
	}
```

**Comment by vinnyparker** [comment]
isn't better prevent user to access other users data? if this data didnt show up for them, so problem solved.

**Comment by JulyanoF** [comment]
Yes. But I did it on listing, saving, access edit page and exclusion (he could access exclusion and edit page changing id in URL if I prevent only on listing)
